src/workerd/api/tests/abortsignal-test.js

```diff
@@ -188,3 +188,30 @@ export const onabortPrototypeProperty = {
     strictEqual(ac.signal.onabort, handler);
   },
 };
+
+// Create a global AbortController instance at module scope to verify
+// that it's possible to create AbortControllers in the global scope
+// This test validates that the fix for https://github.com/cloudflare/workerd/issues/3657 works properly
+const globalAbortController = new AbortController();
```

Comment:
There are a number of issues with this that make this a bit iffy... While this does allow creating the `AbortController` at the global scope, it will end up binding it to the first `IoContext` that accesses it, so any attempt access it from other requests will fail, and any attempt to access it at the global scope will fail. Because of that I'm really not sure if this change is worthwhile. Interested in @kentonv's thoughts.

Reply:
According to AI: 

A better approach might be to:

  1. Completely decouple the AbortSignal's "aborted" state from the IoContext
  2. Only require IoContext when the signal is actually being used to abort I/O operations
  3. Keep track of abort state using simple boolean flags and reason variables
  4. Create the IoContext-dependent canceler only when the signal is attached to I/O

  This would allow AbortController/AbortSignal to be fully usable in global scope for its core functionality, and only throw
  exceptions when actually trying to integrate with I/O operations.